### PR TITLE
Update azdo proxy to latest version

### PR DESCRIPTION
**Pull request by phillebaba**
This change updates azdo proxy to the latest version. The major change is that token generation is no longer done by Terraform but rather in the app.

### CODE REVIEW

File: modules/kubernetes/fluxcd-v2-azdo/templates/azdo-proxy-values.yaml.tpl [hunk 1]
```diff
@@ -1,6 +1,5 @@
-image:
-  tag: v0.3.6
```
Comment (NissesSenap) [inline modules/kubernetes/fluxcd-v2-azdo/templates/azdo-proxy-values.yaml.tpl:2]: Why not define image.tag in the values file any more?

File: modules/kubernetes/fluxcd-v2-azdo/main.tf [hunk 2]
```diff
@@ -99,11 +97,6 @@ data "azuredevops_git_repository" "cluster" {
   name       = var.cluster_repo
 }
 
-resource "random_password" "cluster" {
-  length  = 32
```
Comment (NissesSenap) [inline modules/kubernetes/fluxcd-v2-azdo/main.tf:103]: random_password is set by the application now a days if i remember correctly

File: modules/kubernetes/fluxcd-v2-azdo/templates/azdo-proxy-values.yaml.tpl (hunk 1 above)
Comment (phillebaba) [reply modules/kubernetes/fluxcd-v2-azdo/templates/azdo-proxy-values.yaml.tpl:2]: This was wrong from the beginning and should be picked up by the application/chart version instead.

File: modules/kubernetes/fluxcd-v2-azdo/main.tf (hunk 2 above)
Comment (phillebaba) [reply modules/kubernetes/fluxcd-v2-azdo/main.tf:103]: Yes, all password generation has been removed from the Terraform now.

**Review by simongottschlag — APPROVED** [review]
LGTM :)

**Pull request merged**